wagtail/documents/migrations/0011_add_choose_permissions.py

```diff
@@ -0,0 +1,92 @@
+# Generated by Django 3.1.2 on 2020-10-15 00:52
+
+from django.db import migrations
+
+
+def add_choose_permission_to_admin_groups(apps, _schema_editor):
+    ContentType = apps.get_model('contenttypes.ContentType')
+    Permission = apps.get_model('auth.Permission')
+    Group = apps.get_model('auth.Group')
+
+    # Get document content type
+    document_content_type, _created = ContentType.objects.get_or_create(
+        model='document',
+        app_label='wagtaildocs'
+    )
+
+    # Create the Choose permission (if it doesn't already exist)
+    choose_document_permission, _created = Permission.objects.get_or_create(
+        content_type=document_content_type,
+        codename='choose_document',
+        defaults={'name': 'Can choose document'}
+    )
+
+    # Assign it to all groups which have "Access the Wagtail admin" permission.
+    # This emulates the previous behavior, where everyone who would access the admin
+    # could choose any document in any Collection, because choosing wasn't permissioned.
+    for group in Group.objects.filter(permissions__codename='access_admin'):
+        group.permissions.add(choose_document_permission)
+
+
+def remove_choose_permission(apps, _schema_editor):
+    """Reverse the above additions of permissions."""
+    ContentType = apps.get_model('contenttypes.ContentType')
+    Permission = apps.get_model('auth.Permission')
+    document_content_type = ContentType.objects.get(
+        model='document',
+        app_label='wagtaildocs',
+    )
+    # This cascades to Group
+    Permission.objects.filter(
+        content_type=document_content_type,
+        codename='choose_document'
+    ).delete()
```

Comment:
It's fine to create the permission to use in the data migration (as usually, it wouldn't be created until Django's 'post_migrate' signal is emitted), but I don't think we want to DELETE it here. We're better off letting Django take care of that.

Reply:
I just copied what the images and documents apps do with the original `add_document`, `change_image`, etc. permissions in their respective backwards migrations. See `wagtail/images/migrations/0001_squashed_0021.py` and `wagtail/documents/migrations/0002_initial_data.py`. I wrote this PR's migration based on what those two files do.

As for "letting Django take care of that", what do you mean? In my experience, Django is notorious for *not* deleting permissions for models that have been removed from the database.